File: backend/sim-backend/src/main/java/com/sim/backend/quote/service/QuoteService.java
package com.sim.backend.quote.service;

import com.sim.backend.quote.dto.QuoteRequest;
import com.sim.backend.quote.dto.QuoteResponse;
import com.sim.backend.quote.dto.QuoteUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface QuoteService {

    QuoteResponse create(QuoteRequest request);

    QuoteResponse findById(UUID id);

    List<QuoteResponse> findAll();

    QuoteResponse cancel(UUID id);

    QuoteResponse update(UUID id, QuoteUpdateRequest request);
}

File: frontend/src/app/features/quotes/quote.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import {
  Quote,
  QuoteMaterial,
  QuoteRequest,
  QuoteUpdateRequest
} from '../../shared/models/quote';

@Injectable({
  providedIn: 'root'
})
export class QuoteService {

  private readonly http = inject(HttpClient);

  private readonly apiUrl =
    'http://localhost:8080/api/quotes';

  findAll(): Observable<Quote[]> {
    return this.http.get<Quote[]>(
      this.apiUrl
    );
  }

  findById(id: string): Observable<Quote> {
    return this.http.get<Quote>(
      `${this.apiUrl}/${id}`
    );
  }

  create(request: QuoteRequest): Observable<Quote> {
    return this.http.post<Quote>(
      this.apiUrl,
      request
    );
  }

  update(
    id: string,
    request: QuoteUpdateRequest
  ): Observable<Quote> {

    return this.http.put<Quote>(
      `${this.apiUrl}/${id}`,
      request
    );
  }

  cancel(id: string): Observable<Quote> {
    return this.http.patch<Quote>(
      `${this.apiUrl}/${id}/cancel`,
      {}
    );
  }

  getRequiredMaterials(
    id: string
  ): Observable<QuoteMaterial[]> {

    return this.http.get<QuoteMaterial[]>(
      `${this.apiUrl}/${id}/materials`
    );
  }
}